walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,78 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
 
+/** Finds options for times to schedule a meeting request */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // no options for too long of a request
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration())
+    {
+        Collection<TimeRange> noTimes = Arrays.asList();
+        return noTimes;
+    }
+
+    Collection<String> attendees = new HashSet<>();
+    attendees = request.getAttendees();
```

Comment:
This can be one line, `Collection<String> attendees = request.getAttendees();`

Reply:
Got it.